sklearn/metrics/_classification.py

```diff
@@ -1787,10 +1801,11 @@ def precision_recall_fscore_support(
     pred_sum = tp_sum + MCM[:, 0, 1]
     true_sum = tp_sum + MCM[:, 1, 0]
 
+    xp, _ = get_namespace(y_true, y_pred)
     if average == "micro":
-        tp_sum = np.array([tp_sum.sum()])
-        pred_sum = np.array([pred_sum.sum()])
-        true_sum = np.array([true_sum.sum()])
+        tp_sum = xp.reshape(xp.asarray(xp.sum(tp_sum)), (-1,))
+        pred_sum = xp.reshape(xp.asarray(xp.sum(pred_sum)), (-1,))
+        true_sum = xp.reshape(xp.asarray(xp.sum(true_sum)), (-1,))
```

Comment:
Wouldn't the following work?

```suggestion
        tp_sum = xp.asarray([xp.sum(tp_sum)])
        pred_sum = xp.asarray([xp.sum(pred_sum)])
        true_sum = xp.asarray([xp.sum(true_sum)])
```

Alternatively I would prefer replacing the `(-1,)` shapes by `(1,)` to be more explicit.

Reply:
cupy.array_api fails with the provided suggestion. The complete stack trace is present in the details

<details>

```console
_ test_array_api_compliance[f1_score-check_array_api_multiclass_classification_metric-cupy.array_api-None-None] _

metric = <function f1_score at 0x7bbd4f374d30>
array_namespace = 'cupy.array_api', device = None, dtype_name = None
check_func = <function check_array_api_multiclass_classification_metric at 0x7bbd4f3ce3b0>

    @pytest.mark.parametrize(
        "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
    )
    @pytest.mark.parametrize("metric, check_func", yield_metric_checker_combinations())
    def test_array_api_compliance(metric, array_namespace, device, dtype_name, check_func):
>       check_func(metric, array_namespace, device, dtype_name)

sklearn/metrics/tests/test_common.py:1967: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sklearn/metrics/tests/test_common.py:1812: in check_array_api_multiclass_classification_metric
    check_array_api_metric(
sklearn/metrics/tests/test_common.py:1762: in check_array_api_metric
    metric_xp = metric(a_xp, b_xp, **metric_kwargs)
sklearn/utils/_param_validation.py:213: in wrapper
    return func(*args, **kwargs)
sklearn/metrics/_classification.py:1291: in f1_score
    return fbeta_score(
sklearn/utils/_param_validation.py:186: in wrapper
    return func(*args, **kwargs)
sklearn/metrics/_classification.py:1483: in fbeta_score
    _, _, f, _ = precision_recall_fscore_support(
sklearn/utils/_param_validation.py:186: in wrapper
    return func(*args, **kwargs)
sklearn/metrics/_classification.py:1806: in precision_recall_fscore_support
    tp_sum = xp.asarray([xp.sum(tp_sum)])
/opt/conda/lib/python3.10/site-packages/cupy/array_api/_creation_functions.py:83: in asarray
    res = np.asarray(obj, dtype=dtype)
/opt/conda/lib/python3.10/site-packages/cupy/_creation/from_data.py:88: in asarray
    return _core.array(a, dtype, False, order, blocking=blocking)
cupy/_core/core.pyx:2379: in cupy._core.core.array
    ???
cupy/_core/core.pyx:2402: in cupy._core.core.array
    ???
cupy/_core/core.pyx:2465: in cupy._core.core._array_from_nested_sequence
    ???
cupy/_core/core.pyx:2521: in cupy._core.core._array_from_nested_cupy_sequence
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = Array(2, dtype=int64), axis = (0,)

    def expand_dims(a, axis):
        """Expands given arrays.
    
        Args:
            a (cupy.ndarray): Array to be expanded.
            axis (int): Position where new axis is to be inserted.
    
        Returns:
            cupy.ndarray: The number of dimensions is one greater than that of
            the input array.
    
        .. seealso:: :func:`numpy.expand_dims`
    
        """
        if type(axis) not in (tuple, list):
            axis = axis,
>       return _manipulation._expand_dims(a, axis)
E       TypeError: Argument 'a' has incorrect type (expected cupy._core.core._ndarray_base, got Array)


````

</details>

I think we can go for reshape by (1,)